[torch][ao] Use returned model from Quantizer.transform_for_annotation in prepare_pt2e

dulinriley opened the pull request:
Summary:
The Quantizer subclass can return a new model from `transform_for_annotation`,
and this is common if it uses any ExportPass subclass which does not mutate in-place.

Use the returned model instead of assuming its the same.

Differential Revision: D60869676

jerryzh168 commented:
maybe also add a test?

jerryzh168 commented:
here is a previous test: https://github.com/pytorch/pytorch/blob/32a284c27549449c19e3b1455f531692439d6316/test/quantization/pt2e/test_quantize_pt2e.py#L1498

dulinriley committed:
[torch][ao] Use returned model from Quantizer.transform_for_annotation in prepare_pt2e (#132893)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/132893

The Quantizer subclass can return a new model from `transform_for_annotation`,
and this is common if it uses any ExportPass subclass which does not mutate in-place.

Use the returned model instead of assuming its the same.

Test Plan:
Updated a unit test to return a new graph instead of modifying the
existing one.

Reviewed By: jerryzh168

Differential Revision: D60869676
